views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js

```diff
@@ -53,260 +61,429 @@ define([
         this.initResponseChangeEventListener();
-        self.correctResponses = this.getExistingCorrectAnswerOptions();
-        $responseForm.html(addAnswerOptionBtn());
-        this.initAddAnswerButton();
-        this.renderForm(self.correctResponses);
-    }
-
-    MathEntryInteractionStateResponse.prototype.initAddAnswerButton = function initAddAnswerButton() {
-        var self = this,
-            interaction = self.widget.element,
-            $responseForm = self.widget.$responseForm,
-            $addAnswerBtn = $responseForm.find($('.add-answer-option'));
-
-        $addAnswerBtn.on('click', function () {
-            var newCorrectAnswer;
-
-            if (self.inGapMode() === true) {
-                self.emptyGapFields();
-                var gapExpression = interaction.prop('gapExpression');
-                var gapCount = (gapExpression.match(/\\taoGap/g) || []).length;
+
+        $responseForm.html(responseFormTpl({
+            identifier: interaction.attr('responseIdentifier'),
+            serial: response.serial,
+            min: interaction.prop('min'),
+            max: interaction.prop('max'),
+            mappingDisabled,
+            defaultValue: response.getMappingAttribute('defaultValue')
+        }));
+
+        minMaxComponentFactory($responseForm.find('.response-mapping-attributes > .min-max-panel'), {
+            min: {
+                fieldName: 'lowerBound',
+                value: _.parseInt(response.getMappingAttribute('lowerBound')) || 0,
+                helpMessage: __('Minimal score for this interaction.')
+            },
+            max: {
+                fieldName: 'upperBound',
+                value: _.parseInt(response.getMappingAttribute('upperBound')) || 0,
+                helpMessage: __('Maximal score for this interaction.')
+            },
+            upperThreshold: Number.MAX_SAFE_INTEGER,
+            syncValues: true
+        });
+        this.createScoreResponse();
+        this.initResponseForm();
+        this.initEditingOptions();
+        this.initAlternativeInput();
+
+        // show tooltip
+        tooltip.lookup($responseForm);
+    };
+
+    MathEntryInteractionStateResponse.prototype.initResponseForm = function initResponseForm() {
+        const interaction = this.widget.element;
+
+        let newCorrectAnswer;
+
+        // get first id
+        const $input = this.widget.$container.find('.math-entry-input');
+        const $score = this.widget.$container.find('.math-entry-response-wrap .math-entry-score-input');
+
+        let id = this.correctResponses.getFirstIndex();
+        $score[0].dataset.for = id;
+        $input[0].dataset.index = id;
+
+        let existingReponses = this.getExistingCorrectAnswerOptions();
+        if (existingReponses.length) {
+            const response = interaction.getResponseDeclaration();
+            const mapEntries = response.getMapEntries();
+            existingReponses.forEach((entry, index) => {
```

Comment:
**thought:** I wonder if this is not somehow wrong. The list is filled from the responses, but I don't see it cleared before...

Reply:
It is clear on the `initGlobalVariable`.
I store the first input that is created on the MathEntryInteraction file, then clean the list, and add again this first input.
I must keep the first input because it is linked to the MathQuill API.